sensor/AdblockPlugin.js

```diff
@@ -147,9 +190,142 @@ class AdblockPlugin extends Sensor {
         log.error("Got error when applying adblock policy", err);
       }
     }
+    _scheduleNextReload(oldNextState, curNextState) {
+      if (oldNextState === curNextState) {
+        // no need immediate reload when next state not changed during reloading
+        this.nextReloadFilter.forEach(t => clearTimeout(t));
+        this.nextReloadFilter.length = 0;
+        log.info(`schedule next reload for adblock in ${RELOAD_INTERVAL / 1000}s`);
+        this.nextReloadFilter.push(setTimeout(this._reloadFilter.bind(this), RELOAD_INTERVAL));
+      } else {
+        log.warn(`adblock's next state changed from ${oldNextState} to ${curNextState} during reload, will reload again immediately`);
+        if (this.reloadFilterImmediate) {
+          clearImmediate(this.reloadFilterImmediate)
+        }
+        this.reloadFilterImmediate = setImmediate(this._reloadFilter.bind(this));
+      }
+    }
+
+    async updateFilter() {
+      const config = await this.getAdblockConfig();
+      await this._updateFilter(config);
+    }
+
+    async _updateFilter(config) {
+      for (const key in config) {
+        const configFilePath = `${dnsmasqConfigFolder}/${key}_block.conf`;
```

Comment:
use more specific suffix, e.g., "_adblock.conf", to differentiate with other category block files, e.g., social_block.conf to avoid potential conflict.